### PR TITLE
Disable Jenkins Innerloop Jobs

### PR DESCRIPTION
**Pull request by jashook**
This change will:

1) Remove default PR trigger jobs from being run in jenkins
2) Start submitting tests for linux arm32 in coreclr-ci

It is worth noting that this change will not remove jobs from netci.groovy.
This is to say that it is still possible to trigger innerloop and outerloop
jobs in jenkins, they will just not be run automatically on each new PR.

Past this point the following jobs will be created:

1) Coreclr-ci
2) Windows_NT x64 Formatting
3) Windows_NT x64 Checked CoreFX Tests 
4) Windows_NT x64 Release CoreFX Tests 
5) Ubuntu arm Cross Checked crossgen_comparison Build and Test
6) Windows_NT x86 full_opt ryujit CoreCLR Perf Tests Correctness
7) Windows_NT x86 min_opt ryujit CoreCLR Perf Tests Correctness
8) Windows_NT x64 full_opt ryujit CoreCLR Perf Tests Correctness
9) Windows_NT x64 min_opt ryujit CoreCLR Perf Tests Correctness

Removed jobs (but still explicitly request-able):

1) CentOS7.1 x64 Debug Innerloop Build
2) OSX10.12 x64 Checked Innerloop Build and Test
3) Windows_NT x64 Checked Innerloop Build and Test
4) Windows_NT x64 Checked Innerloop Build and Test (Jit - TieredCompilation=0)
5) Windows_NT x86 Checked Innerloop Build and Test 
6) Windows_NT x86 Checked Innerloop Build and Test (Jit - TieredCompilation=0)
7) Windows_NT arm Debug Innerloop Build
8) Windows_NT arm64 Debug Innerloop Build

To see test results, navigate to the coreclr-ci Azure Dev Ops page at:
https://dev.azure.com/dnceng/public/_build?definitionId=228 and view the build related to your PR.
There is a test tab which will report all test results.

### CODE REVIEW

File: eng/platform-matrix.yml [hunk 1]
```diff
@@ -33,8 +33,7 @@ jobs:
         containerName: ubuntu_1404_arm_cross_build_image
         helixQueues:
           ${{ if eq(variables['System.TeamProject'], 'public') }}:
-            # TODO: add Ubuntu.1404.Arm32.Open once Jenkins has been shutdown
-            asString: ''
+            asString: 'Ubuntu.1404.Arm32'
```
Comment (echesakovMSFT) [inline eng/platform-matrix.yml]: This should be Open

File: netci.groovy [hunk 2]
```diff
@@ -169,13 +169,6 @@ class Constants {
...
     def static prTriggeredValidInnerLoopCombos = [
         'Windows_NT': [
-            'x64': [
-                'Checked'
-            ], 
-            'x86': [
-                'Checked',
-                'Release'
-            ], 
             'arm': [
```
Comment (echesakovMSFT) [inline netci.groovy]: I think you can remove Windows_NT arm Debug - we run this job in AzDO

Comment (jashook) [reply netci.groovy]: Will do.

Comment (echesakovMSFT) [reply netci.groovy]: And same for  Windows_NT arm64 Debug

File: eng/platform-matrix.yml (hunk 1 above)
Comment (echesakovMSFT) [reply eng/platform-matrix.yml]: You also would need to update asArray

**Comment by mmitche** [comment]
@jashook Is there a reason we're not removing the jobs?

**Comment by RussKeldorph** [comment]
@mmitche I think it's mostly just risk given the complexity of netci.groovy.

**Comment by echesakovMSFT** [comment]
@dotnet-bot test ci

**Comment by mmitche** [comment]
@RussKeldorph We still have the git commit history if we remove the jobs. One of the reasons Jenkins has so many issues is that there are so many jobs that are triggered by PR (not just default triggers). The threading model used to process all the comment triggers is extraordinarily expensive.  Removing these is ideal.

In what cases would we actually use or reenable these old jobs or are we just digitally hoarding them?

**Comment by BruceForstall** [comment]
@mmitche IMO, we need more experience with AzDO before removing the jobs entirely from Jenkins. I'm sure that full removal is a short-term goal, but it doesn't need to happen in lock step with disabling the jobs.

**Comment by echesakovMSFT** [comment]
@mmitche How can we disable **Gen CI(dotnet-ci2)** and **Gen CI(dotnet-ci3)** legs triggered by test_ci_command for master/buildpipeline/* ? 

We removed this buildpipeline folder several weeks ago

**Comment by mmitche** [comment]
@echesakov Deleted them from the servers.  Didn't get cleaned up properly.

**Comment by sandreenko** [comment]
I was told that you are enabling arm32/arm64 testing in ADO (currently jobs are named as `Test pri0 Linux arm checked` do not run tests, they only build them), is that true?
So which jobs are now run actual tests in ADO?

**Comment by mmitche** [comment]
I'm not personally.  Maybe @MattGal was?

**Comment by echesakovMSFT** [comment]
@sandreenko Linux arm64 are already enabled, Linux arm32 will be enabled by this PR. To summarize, all AzDO Test jobs except "musl" "rhel" and Windows arm32/arm64 will be building **and** running the tests.

**Comment by jashook** [comment]
@dotnet-bot test ci

**Comment by RussKeldorph** [comment]
>We still have the git commit history if we remove the jobs.

@mmitche I wasn't concerned about the history.  It's just the fact that the groovy is a rats nest of control-flow, so I didn't think it was worth it change more than necessary to avoid risk of breaking something else...along with required debugging, yada yada.

I wasn't aware that just processing the groovy causes excessive load on the server, so that is new information we should consider in the future.

**Comment by mmitche** [comment]
@RussKeldorph It's not the groovy processing that does it, it's that a separate thread is launched for each PR triggered job when a new github comment comes in.  This eventually leads to the machine OOMing.

Anyways, it's fine for now.  Let's just keep full removal on our radar in the near future.

**Comment by jashook** [comment]
@mmitche I have disabled the commit triggers for jobs. @dotnet-bot test ci

**Comment by mmitche** [comment]
@jashook Good to go here?

**Pull request merged**